### PR TITLE
Restrict staff servicing org perms to non-POST requests

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/orgs/tests.py [hunk 1]
```diff
@@ -214,80 +214,37 @@ def test_has_org_perm(self):
             (
                 self.org,
                 "contacts.contact_list",
-                {
-                    self.agent: False,
-                    self.user: True,
-                    self.admin: True,
-                    self.admin2: False,
-                    self.customer_support: True,
```
Comment (rowanseymour) [inline temba/orgs/tests.py:222]: staff servicing perms now happens at view level only and not in `User.has_org_perm` which this is testing

File: temba/api/models.py [hunk 2]
```diff
@@ -72,17 +72,17 @@ def has_permission(self, request, view):
 
         if request.auth:
             # auth token was used
-            role = org.get_user_role(request.auth.user)
+            role = org.get_user_role(request.user)
 
-            # only editors and administrators can use API tokens
-            if role not in APIToken.ALLOWED_ROLES:
+            # only editors, administrators and servicing staff can use API tokens
+            if role not in APIToken.ALLOWED_ROLES and not request.user.is_staff:
```
Comment (rowanseymour) [inline temba/api/models.py:78]: because `org.get_user_role` no longer fakes an Admin role for staff

Comment (rowanseymour) [reply temba/api/models.py:78]: @ericnewcomer @norkans7 we should maybe stop relying on staff API tokens for dashboards.. we could have a special U-Report account which we add to workspaces instead.

File: temba/orgs/views/mixins.py [hunk 3]
```diff
@@ -22,31 +22,33 @@ def get_user(self):
...
+
+        user = self.get_user()
+
+        # check special cases
+        if user.is_anonymous:
+            return False
+        if user.is_superuser:
+            return True
+        if user.is_staff and self.request.method != "POST":
+            return True
```
Comment (ericnewcomer) [inline temba/orgs/views/mixins.py:39]: Maybe better as `self.request.method == "GET"`. I don't think we use PUT or DELETE anywhere, but maybe someday we might?

File: temba/orgs/views/views.py [hunk 4]
```diff
@@ -2143,13 +2144,12 @@ def form_valid(self, form):
...
-
+        @property
+        def permission(self):
             if self.request.META.get("HTTP_X_REQUESTED_WITH") == "XMLHttpRequest" and self.request.method == "GET":
-                perm = "orgs.org_languages"
-
-            return self.request.user.has_perm(perm) or self.has_org_perm(perm)
+                return "orgs.org_languages"
+            else:
+                return "orgs.org_country"
```
Comment (ericnewcomer) [inline temba/orgs/views/views.py:2152]: This is some crazy stuff..

**Comment by ericnewcomer** [comment]
So we are letting staff hit mutable endpoints on the API? Or am I missing something?

**Pull request merged**